test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightPropertySerializerTests.cs

```diff
@@ -267,6 +269,23 @@ public void WritingDatePropertyShouldWorkJsonLight()
         }
         #endregion
 
+        public static IEnumerable<object[]> PrimitiveData => new List<object[]>
+        {
+            new object[] { 42,                     "{\"PrimitiveProperty@odata.type\":\"#Int32\",\"PrimitiveProperty\":42}" },
+            new object[] { new Date(2018, 11, 28), "{\"PrimitiveProperty@odata.type\":\"#Date\",\"PrimitiveProperty\":\"2018-11-28\"}" },
+            new object[] { 8.9,                    "{\"PrimitiveProperty@odata.type\":\"#Double\",\"PrimitiveProperty\":8.9}" },
+            new object[] { true,                   "{\"PrimitiveProperty@odata.type\":\"#Boolean\",\"PrimitiveProperty\":true}" }
+        };
+
+        [Theory]
+        [MemberData("PrimitiveData")]
+        public void WritingEdmPrimitiveTypePropertyShouldWork(object value, string expect)
+        {
+            var primitiveTypeProperty = new ODataProperty { Name = "PrimitiveProperty", Value = value };
+            string actual = this.SerializeProperty(this.entityType, primitiveTypeProperty);
+            Assert.Equal(expect, actual);
+        }

```

Comment:
Should we have a similar ReadingEdmPrimitiveTypePropertyShouldWork in the ODataJsonLightPropertyDeserializer?

Reply:
I added the similar test in reading scenario. Thanks.